pkg/generate/app/cmd/newapp.go

```diff
@@ -315,17 +316,13 @@ func (c *AppConfig) componentsForRepos(repositories app.SourceRepositories) (app
 			errs = append(errs, fmt.Errorf("source not detected for repository %q", repo))
 			continue
 		case info.Dockerfile != nil && (len(c.Strategy) == 0 || c.Strategy == "docker"):
-			df, err := info.Dockerfile.Dockerfile()
-			if err != nil {
-				// an unparseable docker file should not terminate the entire create - it may be valid to Docker but not to us
-				fmt.Fprintf(c.ErrOut, "WARNING: The Dockerfile for the repository %q could not be parsed - no image stream can be created for the FROM\n", info.Path)
+			node := info.Dockerfile.AST()
+			baseImage := dockerfileutil.LastBaseImage(node)
+			if baseImage == "" {
+				errs = append(errs, fmt.Errorf("the Dockerfile in the repository %q has no FROM instruction", info.Path))
 				continue
```

Comment:
\<rant\>Completing the list of crazy possibilities, I think today we don't handle "replacing" the FROM of images that have no FROM instruction. We could certainly insert one at the very beginning of the Dockerfile in that case.
Anyway, I wouldn't recommend anyone that pattern, as you'd be creating invalid Dockerfiles that happen to work in OpenShift, sometimes. Also, if one day Docker will have code to validate the parsed file independent of executing a build, we could use that validation ourselves before manipulating the AST, which would make those crazy Dockerfiles broken. \</rant\>